src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/constrainedtransfer/ConstrainedBoardingSearch.java

```diff
@@ -90,12 +85,59 @@ public boolean transferExist(int targetStopPos) {
         );
     }
 
-    private Collection<ConstrainedTransfer> findMatchingTransfers(
-            Trip sourceTrip,
-            int sourceStopPos
+    private List<TransferForPattern> findMatchingTransfers(
+            TripSchedule tripSchedule,
+            int stopIndex
     ) {
+        final Trip trip = tripSchedule.getOriginalTripTimes().getTrip();
         return currentTransfers.stream()
-                .filter(tx -> translator.source(tx).matches(sourceTrip, sourceStopPos))
+                .filter(t -> t.matchesSourcePoint(stopIndex, trip))
                 .collect(Collectors.toList());
     }
+
+    /**
+     * Find the trip to board (trip index) and the transfer constraint
+     */
+    public T2<Integer, RaptorTransferConstraint> findTimetableTripInfo(
+            RaptorTimeTable<TripSchedule> timetable,
+            List<TransferForPattern> transfers,
+            int stopPos,
+            int sourceTime
+    ) {
+        // Abort after 6 hours
```

Comment:
?

Reply:
Abort after 5 trips past the wanted boarding time.